lib/Runtime/Library/JavascriptProxy.h

```diff
@@ -54,6 +54,19 @@ namespace Js
         static BOOL Is(_In_ RecyclableObject* obj);
         static JavascriptProxy* FromVar(Var obj) { AssertOrFailFast(Is(obj)); return static_cast<JavascriptProxy*>(obj); }
         static JavascriptProxy* UnsafeFromVar(Var obj) { Assert(Is(obj)); return static_cast<JavascriptProxy*>(obj); }
+
+        // before recursively calling something on 'target' use this helper in case there is nesting of proxies.
+        // the proxies could be deep nested and cause SO when processed recursively.
+        static const JavascriptProxy* UnwrapNestedProxies(const JavascriptProxy* proxy)
+        {
+            while (proxy->handler != nullptr && JavascriptProxy::Is(proxy->target))
```

Comment:
could you use TryFromVar<JavascriptProxy> instead of Is/FromVar ?

Reply:
you can see the usage at : JavascriptOperators::IsConstructor
may be refactor and use one of them.

---
In reply to: [205552057](https://github.com/Microsoft/ChakraCore/pull/5529#discussion_r205552057) [](ancestors = 205552057)

Reply:
refactored. 

---
In reply to: [205552555](https://github.com/Microsoft/ChakraCore/pull/5529#discussion_r205552555) [](ancestors = 205552555,205552057)